onnxruntime/test/opaque_api/test_opaque_api.cc

```diff
@@ -266,6 +266,10 @@ TEST_F(OpaqueApiTest, RunModelWithOpaqueInputOutput) {
     str_tensor_value.GetStringTensorContent(actual_result_string.get(), str_len, &offset, 1);
     actual_result_string[str_len] = 0;
     ASSERT_EQ(expected_output.compare(actual_result_string.get()), 0);
+
+    // Free memory
+    allocator.Free(static_cast<void*>(const_cast<char*>(input_names[0])));
+    allocator.Free(static_cast<void*>(const_cast<char*>(output_names[0])));
   } catch (const std::exception& ex) {
```

Comment:
It would be better to change input_name and output_name to be non-const char*s to avoid the casts here. It's also unnecessarily indirect to free using inputs_names[0]/output_names[0]

    char* input_name = session.GetInputName(0, allocator);
    ....
    char* output_name = session.GetOutputName(0, allocator);

    // Free memory
    allocator.Free(input_name);
    allocator.Free(output_name);

Reply:
I think it would be better to keep const char*，avoid changing values. 
It's ok to use input_name/output_name in allocator.Free

Reply:
GetInputName/GetOutputName return 'char *'. Adding const-ness to a local variable and then removing it with const_cast doesn't make sense to me as usage of const_cast generally is a design smell. 

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es50-dont-cast-away-const

Using 'const char*' in inputs_names[] and output_names[] however doesn't need to change.